src/relay/backend/contrib/tensorrt/trt_builder.cc

```diff
@@ -0,0 +1,392 @@
+        {"nn.softmax", new SoftmaxOpConverter()},
+        {"nn.conv2d", new Conv2DOpConverter()},
+        {"nn.dense", new DenseOpConverter()},
+        {"nn.bias_add", new BiasAddOpConverter()},
+        {"add", new ElementWiseBinaryOpConverter()},
+        {"subtract", new ElementWiseBinaryOpConverter()},
+        {"multiply", new ElementWiseBinaryOpConverter()},
+        {"divide", new ElementWiseBinaryOpConverter()},
+        {"power", new ElementWiseBinaryOpConverter()},
+        {"nn.max_pool2d", new PoolingOpConverter()},
+        {"nn.avg_pool2d", new PoolingOpConverter()},
+        {"nn.global_max_pool2d", new GlobalPoolingOpConverter()},
+        {"nn.global_avg_pool2d", new GlobalPoolingOpConverter()},
+        {"exp", new UnaryOpConverter()},
+        {"log", new UnaryOpConverter()},
+        {"sqrt", new UnaryOpConverter()},
+        {"abs", new UnaryOpConverter()},
+        {"negative", new UnaryOpConverter()},
+        {"nn.batch_flatten", new BatchFlattenOpConverter()},
+        {"expand_dims", new ExpandDimsOpConverter()},
+        {"squeeze", new SqueezeOpConverter()},
+        {"concatenate", new ConcatOpConverter()},
+        {"nn.conv2d_transpose", new Conv2DTransposeOpConverter()},
+        {"transpose", new TransposeOpConverter()},
+        {"reshape", new ReshapeOpConverter()},
+        {"nn.pad", new PadOpConverter()},
+        {"sum", new ReduceOpConverter()},
+        {"prod", new ReduceOpConverter()},
+        {"max", new ReduceOpConverter()},
+        {"min", new ReduceOpConverter()},
+        {"mean", new ReduceOpConverter()},
+        {"contrib.adaptive_max_pool2d", new AdaptivePoolingOpConverter()},
+        {"contrib.adaptive_avg_pool2d", new AdaptivePoolingOpConverter()},
+#if TRT_VERSION_GE(5, 1, 5)
+        {"clip", new ActivationOpConverter()},
+        {"nn.leaky_relu", new ActivationOpConverter()},
+        {"sin", new UnaryOpConverter()},
+        {"cos", new UnaryOpConverter()},
+        {"atan", new UnaryOpConverter()},
+        {"ceil", new UnaryOpConverter()},
+        {"floor", new UnaryOpConverter()},
+        {"strided_slice", new StridedSliceOpConverter()},
+#endif
+};
+
+TrtBuilder::TrtBuilder(const std::vector<DLTensor*>& args)
+    : execution_args_(args) {
+  // Create TRT builder and network.
+  static TensorRTLogger trt_logger;
+  builder_ = nvinfer1::createInferBuilder(trt_logger);
+  const int batch_size = args[0]->shape[0];
+  builder_->setMaxBatchSize(batch_size);
+  const size_t workspace_size =
+      dmlc::GetEnv("TVM_TENSORRT_MAX_WORKSPACE_SIZE", size_t(1) << 31);
+  builder_->setMaxWorkspaceSize(workspace_size);
+  const bool use_fp16 = dmlc::GetEnv("TVM_TENSORRT_USE_FP16", false);
+  builder_->setFp16Mode(use_fp16);
+  network_ = builder_->createNetwork();
+}
+
+TrtEngineAndContext TrtBuilder::BuildEngine(const Expr& expr) {
+  // Process graph and create INetworkDefinition.
+  VisitExpr(expr);
+  // Mark outputs.
+  auto network_outputs = node_output_map_[expr.operator->()];
```

Comment:
we may want to check its existence first, otherwise an entry would be automatically created.